src/main/scala/org/broadinstitute/dsde/firecloud/model/MethodRepository.scala

```diff
@@ -1,49 +1,48 @@
-                            @(ApiModelProperty@field)(dataType = "string", value = "Owner")
-                            owner: Option[String] = None,
-                            @(ApiModelProperty@field)(dataType = "string", value = "Payload")
-                            payload: Option[String] = None,
-                            @(ApiModelProperty@field)(dataType = "string", value = "Excluded Field")
-                            excludedField: Option[String] = None,
-                            @(ApiModelProperty@field)(dataType = "string", value = "Included Field")
-                            includedField: Option[String] = None)
+    namespace: Option[String] = None,
+    name: Option[String] = None,
+    snapshotId: Option[Int] = None,
+    synopsis: Option[String] = None,
+    documentation: Option[String] = None,
+    owner: Option[String] = None,
+    payload: Option[String] = None,
+    excludedField: Option[String] = None,
+    includedField: Option[String] = None
+  )
 
-  @ApiModel(value = "Method")
   case class Method(
-                     @(ApiModelProperty@field)(dataType = "string", value = "Namespace")
-                     namespace: Option[String] = None,
-                     @(ApiModelProperty@field)(dataType = "string", value = "Name")
-                     name: Option[String] = None,
-                     @(ApiModelProperty@field)(dataType = "string", value = "Snapshot Id")
-                     snapshotId: Option[Int] = None,
-                     @(ApiModelProperty@field)(dataType = "string", value = "Synopsis")
-                     synopsis: Option[String] = None,
-                     @(ApiModelProperty@field)(dataType = "string", value = "Owner")
-                     owner: Option[String] = None,
-                     @(ApiModelProperty@field)(dataType = "string", value = "Creation Date")
-                     createDate: Option[String] = None,
-                     @(ApiModelProperty@field)(dataType = "string", value = "Payload")
-                     url: Option[String] = None,
-                     @(ApiModelProperty@field)(dataType = "string", value = "The type of the entity (Task, Workflow, Configuration)")
-                     entityType: Option[String] = None)
+    namespace: Option[String] = None,
+    name: Option[String] = None,
+    snapshotId: Option[Int] = None,
+    synopsis: Option[String] = None,
+    owner: Option[String] = None,
+    createDate: Option[String] = None,
+    url: Option[String] = None,
+    entityType: Option[String] = None
+  )
+
+  // represents a method/config permission as exposed to the user from the orchestration layer
+  case class FireCloudPermission(
+    user: Option[String] = None,
+    role: Option[String] = None
+  ) {
+    //verifies non-empty user and that role is one of READER, OWNER or NO ACCESS
+    def verifyValidUserAndRole = AgoraPermissionHandler.hasValidUserAndRole(this)
+    def toAgoraPermission = AgoraPermissionHandler.toAgoraPermission(this)
```

Comment:
I'm concerned about having these two methods here and referencing `(this)`. I'm pretty sure there's a better approach - but I'll make those comments closer to where it's appropriate.